Relationships/RelationshipResolver.cs

```diff
@@ -70,49 +71,93 @@ public class RelationshipResolver
         // A list of all the mods we're going to install.
         private static readonly ILog log = LogManager.GetLogger(typeof (RelationshipResolver));
         private readonly Dictionary<string, CkanModule> modlist = new Dictionary<string, CkanModule>();
-
-        private readonly List<KeyValuePair<Module, Module>> conflicts =
+        private readonly List<CkanModule> user_requested_mods = new List<CkanModule>();
+        private readonly List<KeyValuePair<Module, Module>> conflicts = 
             new List<KeyValuePair<Module, Module>>();
+        private readonly Dictionary<Module, Relationship> reasons = 
+            new Dictionary<Module, Relationship>(new Module.IdentifierEqualilty());
 
-        private readonly Dictionary<Module, Relationship> reasons = new Dictionary<Module, Relationship>();
         private readonly Registry registry;
         private readonly KSPVersion kspversion;
+        private readonly RelationshipResolverOptions options;
+        private readonly HashSet<Module> installed_modules;
+
+
+        public RelationshipResolver(RelationshipResolverOptions options, Registry registry, KSPVersion kspversion)
```

Comment:
Aaah, it's called by our other constructors. That's good. If that's the only way we want it called, can we mark this as private? Alternatively, if we're happy with users calling this directly (which I am) then a touch of documentation letting developers know how to invoke it would be grand. :)